Config show fullname

thecavemanlogic opened the pull request.

thecavemanlogic committed:
Added bread crumb comment

thecavemanlogic committed:
Bug fix: change format from "str" to "uuid"

thecavemanlogic committed:
Passed in fullname into User constructor

thecavemanlogic committed:
Added fullname prompt and got name from form

thecavemanlogic committed:
Retrieve displayFullname variable from POST

thecavemanlogic committed:
Added displayFullname member to Contest class

thecavemanlogic committed:
Added fullname in JSON and made default value

thecavemanlogic committed:
Added displayFullname option

thecavemanlogic committed:
Added logic for fullname to standings reports

thecavemanlogic committed:
Moved statement after comment

thecavemanlogic committed:
Set displayFullname to None on new contest

sschaub commented on opencontest/contest/models/contest.py:
```diff
@@ -24,6 +24,7 @@ def __init__(self, id=None):
             self.showProblInfoBlocks = details.get("showProblInfoBlocks","Off")
             self.problems = [Problem.get(id) for id in details["problems"]]
             self.tieBreaker = str(details.get("tieBreaker", "")) == "true"
+            self.displayFullname = details["displayFullname"]
```
In model constructors, follow the example of using get() to retrieve the dictionary value rather than a normal dictionary index with []. Using get() is forgiving in cases where the data in the json file does not include the key displayFullname. Using [] causes a crash in those scenarios. For example, when I fired up your change to test it, the system crashed immediately because my contest json didn't have that key. When it doesn't take much effort to do so, it's helpful to make changes backwards compatible with existing open-contest databases.

Also, when making code modifictions, take a look at how existing functionality that is similar in nature is implemented. For example, look at the self.tieBreaker line just above where you added your line. Do you notice how it is an on/off option, also? self.tieBreaker stores this as a boolean value in the model. That helps improve readability and reliability throughout the code, because instead of testing for "off" / "on" string literals elsewhere in the code (where a typo in the string literal would result in erroneous behavior), there's exactly *one place* where a string literal has to be used (the comparison to "true" here). 

So, please revise your approach to follow the existing pattern. Also, it would be great if you could put a comment out to the right of both self.tieBreaker and self.displayFullname as follows:

self.tieBreaker = .....      # True = sample correct breaks ties
self.displayFullname = ...     # True = full names displayed in reports

sschaub reviewed (changes requested):
Nice work on this. Just a couple of minor things. On the http://localhost:8000/users page, could you modify the display so that the full name is displayed on the user cards? Also, please revise the  model per my other comment.

thecavemanlogic committed:
Replaced [] with get() and fixed reading bug

thecavemanlogic committed:
Replaced displayFullname string code with boolean code

thecavemanlogic committed:
Check for True instead of "On"

thecavemanlogic committed:
Display user fullname in User card

thecavemanlogic committed:
Receive displayName as boolean by /editContest

thecavemanlogic commented:
I just replaced the dictionary index with get(), made the self.showFullname a boolean, and added the comments.

If on/off values should be stored as booleans, then should ```self.showProblInfoBlocks``` should also store a boolean value as well? It currently stores information as on/off strings too.

thecavemanlogic commented:
Also, to explain the bug I fixed in [bcc4935](https://github.com/bjucps/open-contest/pull/68/commits/bcc4935731500f7ff6343f9cc9bf63a8ccd18bb9)

When the system would read from contest.json the following line:
```python
self.tieBreaker = str(details.get("tieBreaker", "")) == "true"
```
would always evaluate to `False` because `str()` capitalizes the boolean string. So I added added `.lower()` to fix the problem:
```python3
self.tieBreaker = str(details.get("tieBreaker", "")).lower() == "true"
```

sschaub reviewed (approved):
This looks great!
